src/main/java/com/salesforce/dynamodbv2/mt/mappers/sharedtable/impl/MtAmazonDynamoDbBySharedTable.java

```diff
@@ -390,10 +392,44 @@ public QueryResult query(QueryRequest queryRequest) {
     }
 
     /**
-     * TODO: write Javadoc.
+     * Execute a scan meeting the specs of @{link AmazonDynamoDB}, but scoped to single tenants within a shared table.
+     * If no tenant context is specified, a multi tenant scan is performed over the multi tenant shared table.
+     *
+     * Scans scoped to single tenants is not a performant operation. Multi tenant scans are as bad as scans on any other dynamo
+     * table, ie: not great.
      */
     @Override
     public ScanResult scan(ScanRequest scanRequest) {
+        if (getMtContext().getContextOpt().isEmpty()) {
+            // if we're here, we're doing a multi tenant scan on a shared table
+            Preconditions.checkArgument(mtTables.containsKey(scanRequest.getTableName()), scanRequest.getTableName());
+
+
+            ScanResult scanResult =  getAmazonDynamoDb().scan(scanRequest);
+
+            // given the shared table we're working with,
+            // get the function to map the primary key back to
+            // tuple (tenant, virtual table name, and primary attributes)
+            Function<Map<String, AttributeValue>, FieldValue<?>> fieldMapperFunction =
+                getFieldValueFunction(scanRequest.getTableName());
+            List<Map<String, AttributeValue>> unpackedItems = new ArrayList<>(scanResult.getItems().size());
+            List<String> tenants = new ArrayList<>(scanResult.getItems().size());
+            List<String> virtualTables = new ArrayList<>(scanResult.getItems().size());
+            ScanResult ret = new MtScanResult(scanResult, tenants, virtualTables);
+            for (Map<String, AttributeValue> item : scanResult.getItems()) {
+                // go through each row in the scan, and pull out the tenant and table information from the primary key
+                // to separate attributes in each items map
+                FieldValue virtualFieldKeys = fieldMapperFunction.apply(item);
+                TableMapping tableMappping = getMtContext().withContext(virtualFieldKeys.getContext(),
+                    tableName -> getTableMapping(tableName), virtualFieldKeys.getTableName());
+                Map<String, AttributeValue> unpackedVirtualItem = tableMappping.getItemMapper().reverse(item);
```

Comment:
Is there enough commonality with the loop at line 462 below to have some common code?

Reply:
I gave it a try to pull out duplicated logic into a method, and didn't get more than one line out to the shared method, so gave up.